04_구현/12-7_럭키스트레이트.js

```diff
@@ -0,0 +1,17 @@
+const assert = require("assert");
+
+function solution(N) {
+  const chars = String(N).split("");
+  const h = chars.length / 2;
+  return chars.slice(0, h).reduce((sum, c) => sum + Number(c), 0) ===
```

Comment:
```suggestion
  return chars.slice(0, h).map(Number).reduce((sum, c) => sum + c, 0) ===
```

요렇게 형변환을 한번에 처리한 후 다음에 하는 건 어떨까  한번 제안드려봅니다 ㅎㅎ :sweat_smile:

Reply:
map을 쓰게 되면 for문을 한번 더 쓰게 되는건데 느려지진 않을까요??

Reply:
@hyunhaha  제 생각엔 크게 차이는 없을거 같아요..! 
시간복잡도 개념으로 생각할때 for문 1번이나, for문 2번(이중for문X) 이나 둘다 `O(N)` 이라서  크게 상관없지 않을까 싶기도 하구요 ㅎㅎ

`reduce`와 같은 Array.prototype method를 쓰는 건 for문처럼 명령형이 아니라 선언적으로 작성하겠다는 건데, 
map으로 형변환을 하는 부분을 분리하면,  하나의 함수가 하나의 기능만을 수행하게끔 작성하는것이므로
저는 개인적으로 더 선호하는거 같아요 ㅎㅎ

Reply:
오홍 for문 2번은 크게 차이가 없군요..

"하나의 함수가 하나의 기능만을 수행하게끔 작성하는 것" 잘 알아두고 코드 작성할 때 고려해야겠어요! 감사합니다!